ios/RCTARKitNodes.m

```diff
@@ -329,19 +329,27 @@ - (SCNNode *)getNodeWithId:(NSString *)nodeId {
 - (void)removeNode:(NSString *)nodeId {
     
     SCNNode *node = [self getNodeWithId:nodeId];
+
     if (node) {
-        //NSLog(@"removing node: %@ ", key);
+//        NSLog(@"removing node: %@ ", node);
      
         if(node.parentNode) {
             if(node.light) {
                 // see https://stackoverflow.com/questions/47270056/how-to-remove-a-light-with-shadowmode-deferred-in-scenekit-arkit?noredirect=1#comment81491270_47270056
                 node.hidden = YES;
                 [node removeFromParentNode];
+                return;
             } else {
                 [node removeFromParentNode];
+                return;
```

Comment:
these returns prevent the node from beeing removed from self.nodes:

```
 [self.nodes removeObjectForKey:nodeId];
```

self.nodes should always contain all mounted nodes